tests/utils/utils.go

```diff
@@ -452,19 +459,91 @@ func SignTransaction(tx *types.Transaction, key *ecdsa.PrivateKey, chainID *big.
 func CalculateTxParams(
 	ctx context.Context,
 	subnetInfo SubnetTestInfo,
-	fundedAddress common.Address,
+	sender common.Address,
 ) (*big.Int, *big.Int, uint64) {
 	baseFee, err := subnetInfo.ChainRPCClient.EstimateBaseFee(ctx)
 	Expect(err).Should(BeNil())
 
 	gasTipCap, err := subnetInfo.ChainRPCClient.SuggestGasTipCap(ctx)
 	Expect(err).Should(BeNil())
 
-	nonce, err := subnetInfo.ChainRPCClient.NonceAt(ctx, fundedAddress, nil)
+	nonce, err := subnetInfo.ChainRPCClient.NonceAt(ctx, sender, nil)
 	Expect(err).Should(BeNil())
 
 	gasFeeCap := baseFee.Mul(baseFee, big.NewInt(gasUtils.BaseFeeFactor))
 	gasFeeCap.Add(gasFeeCap, big.NewInt(gasUtils.MaxPriorityFeePerGas))
 
 	return gasFeeCap, gasTipCap, nonce
 }
+
+func PrivateKeyToAddress(k *ecdsa.PrivateKey) common.Address {
+	return crypto.PubkeyToAddress(k.PublicKey)
+}
+
+// Throws a Gomega error if there is a mismatch
+func CheckBalance(ctx context.Context, addr common.Address, expectedBalance *big.Int, wsClient ethclient.Client) {
+	bal, err := wsClient.BalanceAt(ctx, addr, nil)
+	Expect(err).Should(BeNil())
+	ExpectBigEqual(bal, expectedBalance)
+}
+
+func TraceTransaction(ctx context.Context, txHash common.Hash, subnetInfo SubnetTestInfo) string {
```

Comment:
This looks to be unused and can be removed.

Reply:
This is a very useful utility function. I'm going to add it to a couple failure paths in the code to make debugging easier but was waiting till the changes to the testing utils settled down.

Reply:
Sounds good!